### PR TITLE
Small Highlander patch

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
tweak: Highlanders can no longer hide behind chairs and plants.
tweak: Highlanders no longer bleed and are no longer slowed down by damage.
/:cl:

I know, I know. I ded, pls nerf. There was a highlander round last night on Sybil where people kept hiding behind chairs and plants (cough lauser cough) and became unhittable. I hate to force things on people, but that's a little scummy, so I cut it out.

Additionally, highlanders are now immune to the slowdown caused by damage, and don't bleed out. This is to promote people dying their deaths for Scotland and not pathetically bleeding out somewhere, and also makes people who have taken a stray hit still able to fight for their own.

Also changed the `world <<` to `send_to_playing_players` and added a "delayed highlander" button that plays music for around 40 seconds before starting it, for when admins want shenanigans or think it'd be funny to watch people scurry around like ants.

Finally, I cut out some old unused code and changed the claymore's greeting message to reflect its current state.

### CODE REVIEW

**Comment by Supermichael777** [comment]
50 seconds of Scotland the brave should do it

**Comment by PKPenguin321** [comment]
Rip tactical chairs

**Comment by Hatfish** [comment]
Highlanders should also be immune to dismemberment, since it's currently very easy to remove your opponents sword arm to prevent them from fighting back.

**Comment by Xhuis** [comment]
I'll look into that.

File: code/game/objects/items/weapons/weaponry.dm [hunk 1]
```diff
@@ -78,22 +77,32 @@ var/highlander_claymores = 0
...
 		nuke_disk.forceMove(get_turf(src))
 		nuke_disk.visible_message("<span class='warning'>The nuke disk is vulnerable!</span>")
 		nuke_disk = null
+	STOP_PROCESSING(SSobj, src)
 	return ..()
 
+/obj/item/weapon/claymore/highlander/process()
+	if(ishuman(loc))
+		var/mob/living/carbon/human/H = loc
+		loc.layer = 5 //NO HIDING BEHIND PLANTS FOR YOU, DICKWEED
```
Comment (ChangelingRain) [inline code/game/objects/items/weapons/weaponry.dm]: USE THE LAYER DEFINES

File: code/game/objects/items/weapons/weaponry.dm [hunk 2]
```diff
@@ -78,22 +77,32 @@ var/highlander_claymores = 0
...
+		H.bleedsuppress = 1 //AND WE WON'T BLEED OUT LIKE COWARDS
```
Comment (ChangelingRain) [inline code/game/objects/items/weapons/weaponry.dm]: Should this be TRUE? Find out!

**Comment by Supermichael777** [comment]
eventually highlanders will just be simplemobs

File: code/game/objects/items/weapons/weaponry.dm (hunk 2 above)
Comment (Xhuis) [inline code/game/objects/items/weapons/weaponry.dm]: Hell if I know. Different maintainers like different things and it's used interchangeably throughout the entire code.

File: code/game/objects/items/weapons/weaponry.dm (hunk 1 above)
Comment (Xhuis) [inline code/game/objects/items/weapons/weaponry.dm]: I tried, but they weren't working for me. I'll put one in, I guess.

File: code/game/objects/items/weapons/weaponry.dm (hunk 2 above)
Comment (RemieRichards) [inline code/game/objects/items/weapons/weaponry.dm]: Except we all agree TRUE/FALSE are better than 1/0, so.

Comment (Xhuis) [inline code/game/objects/items/weapons/weaponry.dm]: Doesn't matter at this point, I added it.

Comment (RemieRichards) [inline code/game/objects/items/weapons/weaponry.dm]: It does, makes sure you understand.

Comment (Xhuis) [inline code/game/objects/items/weapons/weaponry.dm]: Touche.

**Comment by optimumtact** [comment]
I dont' see why we should snowflake this, you should probably nerf tactical chairs and plants instead

**Comment by Xhuis** [comment]
Those are fun on their own and have a legit use. I still want to let people go full METAL GEAR behind objects smaller than them (I once hid behind a Christmas tree as a nuke op), I just want to stop it here.

**Comment by optimumtact** [comment]
I guess so

**Comment by Xhuis** [comment]
Maybe I should add tactical trees, now that I think about it...

File: code/game/objects/items/weapons/weaponry.dm [hunk 3]
```diff
@@ -78,22 +77,32 @@ var/highlander_claymores = 0
...
 		nuke_disk.forceMove(get_turf(src))
 		nuke_disk.visible_message("<span class='warning'>The nuke disk is vulnerable!</span>")
 		nuke_disk = null
+	STOP_PROCESSING(SSobj, src)
 	return ..()
 
+/obj/item/weapon/claymore/highlander/process()
+	if(ishuman(loc))
+		var/mob/living/carbon/human/H = loc
+		loc.layer = FLY_LAYER //NO HIDING BEHIND PLANTS FOR YOU, DICKWEED
```
Comment (ChangelingRain) [inline code/game/objects/items/weapons/weaponry.dm]: honestly this should be, like, LARGE_MOB_LAYER instead, so it isn't too high.

Comment (Xhuis) [inline code/game/objects/items/weapons/weaponry.dm]: The issue I was having was that chairs and plants have the `ABOVE_ALL_MOBS` layer. This was the smallest one I could use that still made sense.

Comment (ChangelingRain) [inline code/game/objects/items/weapons/weaponry.dm]: Uh, plants are ABOVE_MOB_LAYER, not ABOVE_ALL_MOB_LAYER.
Chairs ARE ABOVE_ALL_MOB_LAYER if they're north-facing and you're buckled to them, but you could change that to ABOVE_MOB_LAYER and it'd make sense/not look bad.

Comment (Xhuis) [inline code/game/objects/items/weapons/weaponry.dm]: That could work.

**Comment by optimumtact** [comment]
Please do the needful

**Comment by Xhuis** [comment]
About to head to bed, send me a memo on Coderbus and I'll see it in the morning.

**Comment by Xhuis** [comment]
Damn it, I forgot all about the needful. I'll do it this afternoon, pinky promise.

**Comment by Xhuis** [comment]
Ha! Told you I'd do it!
Also made claymores luminous so people don't get lost in maintenance.

**Pull request merged**